src/client_side.cc

```diff
@@ -1619,23 +1619,8 @@ clientProcessRequest(ConnStateData *conn, const Http1::RequestParserPointer &hp,
     }
 #endif
 
-    if (internalCheck(request->url.path())) {
-        if (internalHostnameIs(request->url.host()) && request->url.port() == getMyPort()) {
-            debugs(33, 2, "internal URL found: " << request->url.getScheme() << "://" << request->url.authority(true));
-            request->flags.internal = true;
-        } else if (Config.onoff.global_internal_static && internalStaticCheck(request->url.path())) {
-            debugs(33, 2, "internal URL found: " << request->url.getScheme() << "://" << request->url.authority(true) << " (global_internal_static on)");
-            request->url.setScheme(AnyP::PROTO_HTTP, "http");
-            request->url.host(internalHostname());
-            request->url.port(getMyPort());
-            request->flags.internal = true;
-            http->setLogUriToRequestUri();
-        } else
-            debugs(33, 2, "internal URL found: " << request->url.getScheme() << "://" << request->url.authority(true) << " (not this proxy)");
-
-        if (ForSomeCacheManager(request->url.path()))
-            request->flags.disableCacheUse("cache manager URL");
-    }
+    if (internalCheck(request->url.path()))
+        request->applyInternalSettings();
```

Comment:
Instead of making sure every caller tests applyInternalSettings() precondition (and tests its the same way) and then asserting that it was tested, let's move the check into the method itself.

Instead of assuming (inside applyInternalSettings()) that `http` is the first request in the pipeline and then asserting that it is still valid, let's move this code into a ClientHttpRequest method rather than HttpRequest method.

```suggestion
    http->checkForInternalAccess();
```